### PR TITLE
Site Icon: Site Icon didn't work for VIPv2 this patch fixes

### PR DESCRIPTION
**Pull request by enejb**
It makes sure that path to the file is used instead of a URL. props @darthhexx.

### CODE REVIEW

**Comment by jeherve** [comment]
Fixes #1895

**Comment by darthhexx** [comment]
Just noting here that I have tested this patch on the VIPv2 system and it works as expected.

**Comment by samhotchkiss** [comment]
LGTM

File: modules/site-icon/jetpack-site-icon.php [hunk 1]
```diff
@@ -438,6 +438,14 @@ public static function all_done_page() {
 		$dir = wp_upload_dir();
 
 		$site_icon_filename = $image_edit->generate_filename( dechex ( time() ) . 'v' . self::$version . '_site_icon', null, 'png' );
+
+		// If the attachment is a URL, then change it to a local file name to allow us to save and then upload the cropped image
+		$check_url = parse_url( $site_icon_filename );
+		if ( isset( $check_url['host'] ) ) {
+			$upload_dir = wp_upload_dir();
+			$site_icon_filename = $upload_dir['path'] . '/' . basename( $site_icon_filename  );
```
Comment (dereksmart) [inline modules/site-icon/jetpack-site-icon.php]: Extra whitespace at the end of this line

**Comment by dereksmart** [comment]
:+1:!

**Pull request merged**